### PR TITLE
More closure size improvements for NixOS 19.09

### PR DESCRIPTION
**Pull request by ttuegel**
###### Motivation for this change

These are more closure size improvements for `release-19.09` because they aren't relevant toward `master`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by delroth — APPROVED** [review]

File: pkgs/tools/graphics/zbar/default.nix [hunk 1]
```diff
@@ -17,6 +17,8 @@ in stdenv.mkDerivation rec {
     sha256 = "0hlxakpyjg4q9hp7yp3har1n78341b4knwyll28hn48vykg28pza";
   };
 
+  outputs = [ "out" "dev" ];
```
Comment (worldofpeace) [inline pkgs/tools/graphics/zbar/default.nix]: we actually already went a lot further with this
* https://github.com/NixOS/nixpkgs/commit/e1cb839d17d7ec6bdf97624e0f1c0f0af6fb62c3#diff-b15562677b44058bddf61ce5bc1b5c20

There's other commits there that clean up dependencies etc. on master that perhaps we can backport.

Comment (ttuegel) [reply pkgs/tools/graphics/zbar/default.nix]: I wasn't aware of that pull request, thanks! I will backport that instead, as I prefer not to diverge from master.

**Comment by srhb** [comment]
Why was this not relevant for master? This ends up being weird because now we need to fix up eg. kea which broke because of the mariadb, but only in release-19.09. That's not a divergence I'm used to seeing.

**Comment by ttuegel** [comment]
> Why was this not relevant for master? This ends up being weird because now we need to fix up eg. kea which broke because of the mariadb change, but only in release-19.09. That's not a divergence I'm used to seeing.

The relevant parts of this are in master, but the branches had already diverged to the point that I couldn't blindly cherry-pick the changes into the release branch.

**Comment by srhb** [comment]
@ttuegel Thanks! I understand what's going on now. :)

**Pull request merged**